PMM-7307 Release Notes 2.14.0 (#360)
percona/pmm-doc: docs/release-notes/2.14.0.md

# Percona Monitoring and Management 2.14.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">January 26, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## Release Highlights

- Switch to push metrics by default

>     In PMM v2.12.0, Percona replaced its metrics collection service (formerly Prometheus) with VictoriaMetrics. Historically, PMM used a "pull" method with Prometheus. VictoriaMetrics can operate in either a "pull" or "push" method. When PMM v2.12.0 was released, Percona kept the method as "pull". Now with PMM v2.14.0, Percona is shifting the default to "push". This [blog post](https://www.percona.com/blog/2020/12/01/foiled-by-the-firewall-a-tale-of-transition-from-prometheus-to-victoriametrics/) describes the two methods and why "push" benefits its users. Also, [here is a post](https://www.percona.com/blog/2020/12/16/percona-monitoring-and-management-migration-from-prometheus-to-victoriametrics-faq/) by Peter Zaitzev of FAQs relating to the move to VictoriaMetrics and the "push" model. Documentation on the "push" method is [here](../details/victoria-metrics.md).

Reviewer: ```suggestion
    In PMM v2.12.0, Percona replaced its metrics collection service (formerly Prometheus) with VictoriaMetrics. Historically, PMM used a "pull" method with Prometheus while VictoriaMetrics can operate in either a "pull" or "push" method. When PMM v2.12.0 was released, Percona kept the default method as "pull". Now with PMM v2.14.0, Percona is shifting the default to "push" for all newly added instances. This [blog post](https://www.percona.com/blog/2020/12/01/foiled-by-the-firewall-a-tale-of-transition-from-prometheus-to-victoriametrics/) describes the two methods and why "push" benefits its users. Also, [here is a post](https://www.percona.com/blog/2020/12/16/percona-monitoring-and-management-migration-from-prometheus-to-victoriametrics-faq/) by Peter Zaitzev of FAQs relating to the move to VictoriaMetrics and the "push" model. Documentation on the "push" method is [here](../details/victoria-metrics.md).
```

Author: Change made in https://jira.percona.com/browse/RM-884 ready for next jirabot run

> - DBaaS Preview phase 1.0 (Technical  Preview)

Reviewer: Not sure why are we re-announcing it?  Anything new and noteworthy or just keeping it at the forefront of peoples minds?

Author: @rnovikovP Please attend

Reviewer 2: this is from our Internal release announcement.  I would keep it this way for now, because the change is important and it's better to tell about it. Especially if someone will be migrating om 2.10  for example. (they'll not see  RN from the previous release )

    DBaaS enables non-DBAs (software architects, developers, site reliability engineers, etc.) to perform typical DBA tasks to manage an organization’s database environment via user interfaces and automation orchestration.

    **Note**: This capability is "feature-flagged" and turned off by default. Users require a variable to be passed to PMM to expose this functionality.

> - Pass configured SSL certificates to connect PMM for MongoDB

Reviewer: So last release enabled SSL in Mongo, this one allows you to use custom certificates?  This otherwise reads exactly like the previous highlight so I'm confused as to "what's new about it"?

Author: @rnovikovP  Please attend

Reviewer 2: copy-paste error.  removing from RN. (did this in Jira  )

    Enables PMM administrators to setup configured SSL certificate "keys" to authenticate the connection to PMM, specifically for setting up MongoDB. This is a critical security requirement especially in large enterprise infrastructure environments.

- External services presentation on node summary dashboard

>     Improvements to the user experience for adding and viewing external services (any data that can be monitored by a Prometheus export such as: non-Percona supported databases like Reddis, ElasticSearch, Cassandra, etc. or an organization’s external application) on the Node Summary dashboard of PMM.

Reviewer: ```suggestion
    Improvements to the user experience for adding and viewing external services (any data that can be monitored by a Prometheus exporter such as: non-Percona supported databases like Reddis, ElasticSearch, Cassandra, etc. or an organization’s external application) on the Node Summary dashboard of PMM.
```

Author: Change made in https://jira.percona.com/browse/RM-884 ready for next jirabot run

## New Features

> * [PMM-5765](https://jira.percona.com/browse/PMM-5765): Add external serverless service (no host or address) with new 'pmm-admin add external-serverless' option

Reviewer: @rnovikovP this makes no sense (the no host or address) considering you can specify a URL... does this instead mean a PMM server can monitor a remote node that is not running the PMM Client package?

Author: We need to explain what 'serverless' means. It's a new term suddenly appearing and not mentioned in the pmm-admin man page.

Reviewer 2: This need to be marked as  Technical Preview

Author: Add it to the summary and JIRABOT will extract it

Author: Done

Reviewer 2: @PaulJacobs-percona  looks like in Jira the issue  looks good already. can you  check it, pls

> * [PMM-7015](https://jira.percona.com/browse/PMM-7015): Create DB cluster with random password

Reviewer: ```suggestion
* [PMM-7015](https://jira.percona.com/browse/PMM-7015): DBaaS Preview: Create DB cluster with randomly generated password
```

Author: Changes made in https://jira.percona.com/browse/PMM-7015 ready for next JIrabot run

* [PMM-6941](https://jira.percona.com/browse/PMM-6941): Integrated Alerting: Ability to delete an 'Alert Rule Template'

## Improvements

* [PMM-7145](https://jira.percona.com/browse/PMM-7145): 'Push' metrics mode is default when adding services and nodes (All agents collecting data from Services and Nodes will now use PUSH model if not specified explicitly. You will still be able to use '-metrics-mode' flag to use Pull metrics if needed. All previously set up agents will keep their existing mode. To change these you need to remove and re-add them.)

> * [PMM-7282](https://jira.percona.com/browse/PMM-7282): Integrated Alerting: Ability to create rule without channels and filters

Reviewer: ```suggestion
* [PMM-7282](https://jira.percona.com/browse/PMM-7282): Integrated Alerting: Enable ability to create rule without channels and filters
```

Author: Written as a headline, "the" is implied when reading (THE ability to ...). Readers know this is an improvement so "adding", "enabling", "implementing" are implied too. If we start turning headlines into proper grammatical sentences, we will loose their impact. Adding the word 'Enable' doesn't help.

* [PMM-7082](https://jira.percona.com/browse/PMM-7082): Integrated Alerting: Severity levels are color-coded
* [PMM-7065](https://jira.percona.com/browse/PMM-7065): Integrated Alerting: Show rule details for items in Alert Rules list

## Bugs Fixed

> * [PMM-5137](https://jira.percona.com/browse/PMM-5137): Swagger page redirect isn't working

Reviewer: ```suggestion
* [PMM-5137](https://jira.percona.com/browse/PMM-5137): /swagger page redirect to /swagger/ isn't working
```

Author: These details are in the ticket. Reader doesn't benefit from this additional nuance.

* [PMM-7178](https://jira.percona.com/browse/PMM-7178): pg_stat_monitor integration is broken: agent stays in 'Waiting' status
* [PMM-7105](https://jira.percona.com/browse/PMM-7105): Query Analytics: no 'Example' or 'Explain' data for MariaDB
* [PMM-7239](https://jira.percona.com/browse/PMM-7239): Integrated Alerting: Validate Slack channel names in Notification Channels
* [PMM-7213](https://jira.percona.com/browse/PMM-7213): MySQL InnoDB Details dashboard: remove color-coding on 'Data Buffer Pool Fit' element
* [PMM-7167](https://jira.percona.com/browse/PMM-7167): Some panels not visible when using long time intervals (e.g. 30 days)
* [PMM-7133](https://jira.percona.com/browse/PMM-7133): Incorrect descriptions for data links in dashboards
* [PMM-6904](https://jira.percona.com/browse/PMM-6904): 'pmm-admin annotate' command crashes for non-generic node types
* [PMM-6902](https://jira.percona.com/browse/PMM-6902): No query Examples on PostgreSQL 12 with pg_stat_monitor
* [PMM-6838](https://jira.percona.com/browse/PMM-6838): ProxySQL Instance Summary dashboard: Incorrect "Hostgroup Size" formula

> * [PMM-6490](https://jira.percona.com/browse/PMM-6490): Limit of 100 AWS RDS instances

Reviewer: wasn't this a community contribution? don't we thank the contributor for that typically?

Author: Jirabot automagically accredits community contributions. None of these 4 are community.

- DBaaS Preview phase 1.0 (Technical  Preview)

    DBaaS enables non-DBAs (software architects, developers, site reliability engineers, etc.) to perform typical DBA tasks to manage an organization's database environment via user interfaces and automation orchestration.

    **Note**: This capability is feature-flagged and turned off by default. Users require a variable to be passed to PMM to expose this functionality.

- Pass configured SSL certificates to connect PMM for MongoDB

    Enables PMM administrators to setup configured SSL certificate keys to authenticate the connection to PMM, specifically for setting up MongoDB. This is a critical security requirement especially in large enterprise infrastructure environments.

- External services presentation on node summary dashboard

    Improvements to the user experience for adding and viewing external services (any data that can be monitored by a Prometheus exporter such as: non-Percona supported databases like Reddis, ElasticSearch, Cassandra, etc. or an organization's external application) on the Node Summary dashboard of PMM.

## New Features

> * [PMM-5765](https://jira.percona.com/browse/PMM-5765): Add external serverless service (no host or address) with new 'pmm-admin add external-serverless' option

Reviewer 2: ```suggestion
* [PMM-5765](https://jira.percona.com/browse/PMM-5765): Add an ability to monitor External Services where is no ability to install pmm-client. This is done by new command pmm-admin add external-serverless, read more   in  [.....  ]  (Link to the Reference manual for the command )
```

Author: 'Technical Preview' added to ticket's ADDITIONAL RELEASE NOTES TEXT field. Let's NOT link to the man page but just say it's there. Linking is a maintenance overhead and unnecessary for our supersmart readers.

> * [PMM-5765](https://jira.percona.com/browse/PMM-5765): Add external serverless service (no host or address) with new 'pmm-admin add external-serverless' option

Reviewer 2: ```suggestion
* [PMM-5765](https://jira.percona.com/browse/PMM-5765): Add an ability to monitor External Services where is no ability to install pmm-client. This is done by new command pmm-admin add external-serverless, read more   in  [.....  ]  (Link to the Reference manual for the command )
```

* [PMM-7015](https://jira.percona.com/browse/PMM-7015): DBaaS Preview: Create DB cluster with randomly-generated password
* [PMM-6941](https://jira.percona.com/browse/PMM-6941): Integrated Alerting: Ability to delete an 'Alert Rule Template'

## Improvements

* [PMM-7145](https://jira.percona.com/browse/PMM-7145): 'Push' metrics mode is default when adding services and nodes (All agents collecting data from Services and Nodes will now use PUSH model if not specified explicitly. You will still be able to use '-metrics-mode' flag to use Pull metrics if needed. All previously set up agents will keep their existing mode. To change these you need to remove and re-add them.)

> * [PMM-7282](https://jira.percona.com/browse/PMM-7282): Integrated Alerting: Ability to create rule without channels and filters

Reviewer 2: PMM-7282 - is from the feature in Technical Preview (Integrated Alerting) . should we group them and mark them once?

Author: Jirabot can't sort on that (unless you add a flag and adjust the code)

* [PMM-7082](https://jira.percona.com/browse/PMM-7082): Integrated Alerting: Severity levels are color-coded
* [PMM-7065](https://jira.percona.com/browse/PMM-7065): Integrated Alerting: Show rule details for items in Alert Rules list

> * [PMM-7048](https://jira.percona.com/browse/PMM-7048): Move Create Cluster button out of Step 2

Reviewer 2: ```suggestion
* [PMM-7048](https://jira.percona.com/browse/PMM-7048): DBaaS: Simplify Cluster creation by moving Create Cluster button to earlier steps
```

## Bugs Fixed

> * [PMM-7312](https://jira.percona.com/browse/PMM-7312): Error when accessing TimeSeries information panels (with PMM v2.13.0 via Docker, 200+ monitored servers, 2-3 services each)

Reviewer 2: ```suggestion
* [PMM-7312](https://jira.percona.com/browse/PMM-7312): Error when accessing Metrics data on Dashboards for large installations
```

* [PMM-5137](https://jira.percona.com/browse/PMM-5137): Swagger page redirect isn't working
* [PMM-7251](https://jira.percona.com/browse/PMM-7251): Integrated Alerting: Error 'Rule with ID "mysql_version" not found' if both Security Threat Tool and Integrated Alerting enabled

> * [PMM-7247](https://jira.percona.com/browse/PMM-7247): Disk size is always 0 for Percona XtraDB cluster

Reviewer 2: ```suggestion
* [PMM-7247](https://jira.percona.com/browse/PMM-7247): DBaas: Disk size is always 0 for Percona XtraDB cluster
```

> * [PMM-7178](https://jira.percona.com/browse/PMM-7178): pg_stat_monitor integration is broken: agent stays in 'Waiting' status

Reviewer 2: ```suggestion
* [PMM-7178](https://jira.percona.com/browse/PMM-7178): pg_stat_monitor integration is broken with version 0.6.0 of the plugin
```

* [PMM-7105](https://jira.percona.com/browse/PMM-7105): Query Analytics: no 'Example' or 'Explain' data for MariaDB
* [PMM-7239](https://jira.percona.com/browse/PMM-7239): Integrated Alerting: Validate Slack channel names in Notification Channels
* [PMM-7213](https://jira.percona.com/browse/PMM-7213): MySQL InnoDB Details dashboard: remove color-coding on 'Data Buffer Pool Fit' element
* [PMM-7167](https://jira.percona.com/browse/PMM-7167): Some panels not visible when using long time intervals (e.g. 30 days)
* [PMM-7133](https://jira.percona.com/browse/PMM-7133): Incorrect descriptions for data links in dashboards
* [PMM-6904](https://jira.percona.com/browse/PMM-6904): 'pmm-admin annotate' command crashes for non-generic node types
* [PMM-6902](https://jira.percona.com/browse/PMM-6902): No query Examples on PostgreSQL 12 with pg_stat_monitor
* [PMM-6838](https://jira.percona.com/browse/PMM-6838): ProxySQL Instance Summary dashboard: Incorrect "Hostgroup Size" formula

> * [PMM-6490](https://jira.percona.com/browse/PMM-6490): Limit of 100 AWS RDS instances

Reviewer 2: ```suggestion
* [PMM-6490](https://jira.percona.com/browse/PMM-6490): Limit of 100 AWS RDS instances. Thanks, https://github.com/vlinevych for the fix
```

Author: Added to ticket's ADDITIONAL TEXT field

> * [PMM-6096](https://jira.percona.com/browse/PMM-6096): pmm-agent's connection checker does not check authentication

Reviewer 2: ```suggestion
* [PMM-6096](https://jira.percona.com/browse/PMM-6096): pmm-agent's connection checker does not check authentication for MongoDB
```

## Bugs Fixed

* [PMM-7312](https://jira.percona.com/browse/PMM-7312): Error when accessing Metrics data on Dashboards for large installations
* [PMM-7310](https://jira.percona.com/browse/PMM-7310): VictoriaMetrics consuming 100's Gb's of disk in /tmp/searchResults in PMM 2.13.0
* [PMM-5137](https://jira.percona.com/browse/PMM-5137): Swagger page redirect isn't working
* [PMM-7323](https://jira.percona.com/browse/PMM-7323): Removing DB Cluster from wrong Kubernetes Cluster
* [PMM-7251](https://jira.percona.com/browse/PMM-7251): Integrated Alerting: Error 'Rule with ID "mysql_version" not found' if both Security Threat Tool and Integrated Alerting enabled
* [PMM-7247](https://jira.percona.com/browse/PMM-7247): DBaas: Disk size is always 0 for Percona XtraDB cluster
* [PMM-7178](https://jira.percona.com/browse/PMM-7178): pg_stat_monitor integration is broken with version 0.6.0 of the plugin
* [PMM-7169](https://jira.percona.com/browse/PMM-7169): Old data (from Prometheus) not deleted when Retention period expires
* [PMM-7105](https://jira.percona.com/browse/PMM-7105): Query Analytics: no 'Example' or 'Explain' data for MariaDB

> * [PMM-7085](https://jira.percona.com/browse/PMM-7085): DBaaS dashboard showing in list of dashboards even when feature flag not enabled 

Reviewer 2: ```suggestion
```

Reviewer 2: Issue removed from the release
